lib/Sema/TypeCheckProtocol.cpp

```diff
@@ -5391,6 +5391,32 @@ void swift::diagnoseConformanceFailure(Type T,
     return;
   }
 
+  // Special case: a distributed actor conformance often can fail because of
+  // a missing ActorSystem (or DefaultDistributedActorSystem) typealias.
+  // In this case, the "normal" errors are an avalanche of errors related to
+  // missing things in the actor that don't help users diagnose the root problem.
+  // Instead, we want to suggest adding the typealias.
+  if (Proto->isSpecificProtocol(KnownProtocolKind::DistributedActor)) {
+    auto nominal = T->getNominalOrBoundGenericNominal();
+    if (!nominal)
+      return;
+
+    // If it is missing the ActorSystem type, suggest adding it:
+    // FIXME(distributed): use getDistributedActorSystemType(nominal); once merged
+    auto &C = nominal->getASTContext();
+    auto DA = C.getDistributedActorDecl();
+    Type selfType = nominal->getSelfInterfaceType();
+    auto conformance = nominal->getParentModule()->lookupConformance(selfType, DA);
+    auto systemTy = conformance.getTypeWitnessByName(selfType, C.Id_ActorSystem);
```

Comment:
I have this logic in the new DistributedDecl.cpp but ripping it out is a large change coming in a different PR...